python/dgl/backend/pytorch/sparse_emb.py

```diff
@@ -0,0 +1,418 @@
+import torch as th
+import abc
+from abc import abstractmethod
+from .tensor import attach_grad, is_recording
+
+class GraphSparseEmbedding:
+    '''Sparse embeddings for graph.
+
+    DGL provides a sparse embedding to support models that require learnable embeddings.
+    DGL's sparse embeddings are mainly used for learning node embeddings of graph models.
+    The sparse embeddings have to be updated by DGL's optimizers instead of
+    the optimizers provided by the deep learning frameworks (e.g., Pytorch).
+
+    To support efficient training on a graph with many nodes, the embeddings support sparse
+    updates. That is, only the embeddings involved in a mini-batch computation are updated.
+    Currently, DGL provides two optimizer: `SparseAdagrad` and `SparseAdam. DGL will provide more
+    optimizers in the future.
+
+    Parameters
+    ----------
+    num_embeddings : int
+        The number of embeddings. Currently, the number of embeddings has to be the same as
+        the number of nodes or the number of edges.
+    embedding_dim : int
+        The dimension size of embeddings.
+    device : th.device
+        Target device to manipulate embedding related opations
+    name : str
+        The name of the embeddings. The name should uniquely identify embeddings in a system.
+    init_func : callable, optional
+        The function to create the initial data. If the init function is not provided,
+        the values of the embeddings are initialized to zero.
+    rank : int, optional
+        The rank in multi-gpu process group. -1 means single GPU or CPU training. Default: -1.
+    world_size : int, optional
+        Number of devices used in mutli-gpu training. Default: 0.
+    queues : list of multiprocessing.Queue, optional
+        Used by multi-gpu processes to share embedding tensor. Default: None.
+
+    Examples
+    --------
+    Before launching multiple gpu processes
+
+    >>> queues = [mp.Queue() for _ in range(num_gpus)]
+    >>> def initializer(shape, dtype):
+            arr = th.zeros(shape, dtype=dtype)
+            arr.uniform_(-1, 1)
+            return arr
+
+    In each multiple gpu process
+
+    >>> emb = dgl.backend.F.GraphSparseEmbedding(g.number_of_nodes(), 10, init_func=initializer,
```

Comment:
can we just use dgl.GraphSparseEmbedding?

Reply:
We can do that.

Reply:
Done